packages/blocks/src/api/raw-handling/test/paste-handler.js

```diff
@@ -287,4 +287,33 @@ describe( 'pasteHandler', () => {
 		expect( result.name ).toEqual( 'core/video' );
 		expect( result.isValid ).toBeTruthy();
 	} );
+
+	it( 'can handle Grok markdown wrapped in a styled span', () => {
+		// Simulates pasting markdown from Grok, which wraps content in a styled span
+		const grokMarkdownHTML =
```

Comment:
This is not quite what I had in mind. There's a folder with HTML input and output that we use as integration tests. What I had in mind is just adding the the input and expected output to this folder. It can be found at `test/integration/fixtures/documents`.